qiskit-superstaq/qiskit_superstaq/superstaq_backend.py

```diff
@@ -172,17 +172,21 @@ def run(
 
     def retrieve_job(self, job_id: str) -> qss.SuperstaqJob:
         """Gets a job that has been created on the Superstaq API.
-
         Args:
             job_id: The UUID of the job. Jobs are assigned these numbers by the server during the
             creation of the job.
-
         Returns:
             A `qss.SuperstaqJob` which can be queried for status or results.
-
         Raises:
             ~gss.SuperstaqServerException: If there was an error accessing the API.
         """
+        warnings.warn(
```

Comment:
Use Cirq's deprecation infrastructure. [Here's an example](https://github.com/quantumlib/Cirq/blob/d1b0430d27faa75efe933f8dc4bfe1a08b8fed09/cirq-core/cirq/devices/noise_utils.py#L99)

Reply:
Is it worth having our own deprecation warning decorator in `gss`?

Reply:
It is not.

Reply:
What version should we say is the deadline for deprecation?

Reply:
Let's do `0.5.40`. I made it up.

Reply:
Double checking. This is the warning body we will have 

`f'{qualname} was used but is deprecated.\n'
                f'It will be removed in cirq {deadline}.\n'
              f'{fix}\n'`. 
             Given that the message will state that the function will be deprecated for a certain cirq version, should we use it for `qss`?

Reply:
Ah I see. What we have now is fine then.